canonical/ubuntu-pro-client
# Landscape fixes

## Why is this needed?

We adjusted, via the resource definition, the landscape UX to not uninstall landscape-client on disable. That revealed some bugs but also opportunities to simplify the `pro enable landscape` experience overall.

This also fixes the problem where pro was interfering with landscape-clients on releases prior to mantic.

Fixes: #2743
Fixes: #2839 
Fixes: #2840

## Test Steps
Please review landscape.feature to make sure the new UX makes sense.

The most important bug to test is #2743 and there is a new behave test for this, but to manually test:
* launch a jammy container and install this version of pro-client from this ppa https://launchpad.net/~orndorffgrant/+archive/ubuntu/pro-client-landscape-test
* `pro attach`
* `apt install landscape-client`
* `landscape-config`
* `pro detach`
* ensure that landscape-client is still configured and running

## Checklist


## Does this PR require extra reviews?

 - [x] Yes - @mdscunningham or @Monochromics please take a look and let me know if this resolves most of the issues you're seeing. Also @Perfect5th or @silverdrake11 make sure I'm not doing anything wrong in landscape interactions here.
